src/main/java/com/example/Restaurant_Management_System_REST_API/service/InventoryItemService.java

```diff
@@ -0,0 +1,94 @@
+package com.example.Restaurant_Management_System_REST_API.service;
+
+import com.example.Restaurant_Management_System_REST_API.DTO.InventoryItemDTOs.InventoryItemDTORequest;
+import com.example.Restaurant_Management_System_REST_API.DTO.InventoryItemDTOs.InventoryItemDTOResponse;
+import com.example.Restaurant_Management_System_REST_API.exception.NotFoundInDatabaseException;
+import com.example.Restaurant_Management_System_REST_API.model.entity.InventoryItem;
+import com.example.Restaurant_Management_System_REST_API.model.entity.Supplier;
+import com.example.Restaurant_Management_System_REST_API.repository.InventoryItemRepository;
+import com.example.Restaurant_Management_System_REST_API.repository.SupplierRepository;
+import com.example.Restaurant_Management_System_REST_API.service.generic.GenericBasicCrudOperations;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+public class InventoryItemService implements GenericBasicCrudOperations<InventoryItemDTOResponse, InventoryItemDTORequest,
+        Long> {
+
+    private InventoryItemRepository inventoryItemRepository;
+    private SupplierRepository supplierRepository;
+    private ModelMapper modelMapper;
+
+    @Override
+    public InventoryItemDTOResponse create(InventoryItemDTORequest inventoryItemDTORequest) throws NotFoundInDatabaseException {
+
+        //TODO: prepare SupplierController and service and check if below works with existing suppliers
+       //Checking if provided supplier exists in database - if not -> throwing an exception
+        checkIfSupplierExist(inventoryItemDTORequest);
+
+        InventoryItem inventoryItem = modelMapper.map(inventoryItemDTORequest, InventoryItem.class);
+        inventoryItemRepository.save(inventoryItem);
+
+        return modelMapper.map(inventoryItem, InventoryItemDTOResponse.class);
+    }
+
+    private void checkIfSupplierExist(InventoryItemDTORequest inventoryItemDTORequest) throws NotFoundInDatabaseException {
+        if(inventoryItemDTORequest.getSupplier() != null) {
```

Comment:
if the getSupplier() returns null this method should do nothing?

Reply:
exactly! it is not obligatory!